platform/fabric/services/state/transaction.go

```diff
@@ -36,10 +36,10 @@ func Wrap(tx *endorser.Transaction) (*Transaction, error) {
 
 // NewTransaction returns a new instance of a state-based transaction that embeds a single namespace.
 func NewTransaction(context view.Context) (*Transaction, error) {
-	return newTransaction(utils.MustGet(fabric.GetNetworkServiceProvider(context)), context)
+	return NewTransactionWithFNSP(utils.MustGet(fabric.GetNetworkServiceProvider(context)), context)
```

Comment:
please, don't panic here, check what's the output of `fabric.GetNetworkServiceProvider` and return an error.